src/dataflow/src/render/upsert.rs

```diff
@@ -50,6 +50,8 @@ pub(crate) fn upsert<G>(
     stream: &Stream<G, DecodeResult>,
     as_of_frontier: Antichain<Timestamp>,
     operators: &mut Option<LinearOperator>,
+    key_arity: usize,
+    // Full arity, including the key(s)
```

Comment:
Including the **key** or **key columns** not keys.

We don't *quite* follow this consistently in the codebase (I wish we did), but in our terminology "a key" can be multiple columns (or even a series of arbitrary expressions, though in upsert it can only be columns).